src/main/java/org/terasology/lost/LevelSpawnSystem.java

```diff
@@ -62,15 +62,15 @@
     public void onBiomeChange(OnBiomeChangedEvent event, EntityRef player,
                               ProgressTrackingComponent progressTrackingComponent) {
         LocationComponent loc = player.getComponent(LocationComponent.class);
-        Vector3f playerLocation = loc.getWorldPosition();
-        playerLocation = playerLocation.add(loc.getWorldDirection().scale(3));
+        Vector3f playerLocation = loc.getWorldPosition(new Vector3f());
+        playerLocation = playerLocation.add(loc.getWorldDirection(new Vector3f()).mul(3));
         // nearby area radius for which facets are fetched
         int searchRadius = 400;
 
         // create Region to be searched
         Vector3i extent = new Vector3i(searchRadius, 1, searchRadius);
-        Vector3i desiredPos = new Vector3i(playerLocation.getX(), 1, playerLocation.getZ());
-        Region3i searchRegion = Region3i.createFromCenterExtents(desiredPos, extent);
+        Vector3i desiredPos = new Vector3i(new Vector3f(playerLocation.x(), 1, playerLocation.z()), RoundingMode.FLOOR);
+        BlockRegion searchRegion = new BlockRegion(desiredPos).expand(extent);
```

Comment:
I think we can keep this like it is. Just a note on performance:

The search radius is static, and therefore the `extent` is also the same all the time. We could create a region of the desired size once in the system, and only translate it to the `desiredPos` when the event occurs.
I doubt this is the bottleneck here, though.